server/src/main/java/org/elasticsearch/index/mapper/BlockSourceReader.java

```diff
@@ -131,6 +139,24 @@ public RowStrideReader rowStrideReader(LeafReaderContext context) {
         }
     }
 
+    public static class GeometriesBlockLoader extends SourceBlockLoader {
```

Comment:
It makes me feel uneasy, cannot we reuse the BytesRef  reader as we are reading an array of bytes?

Reply:
Right now we need this to get to the code in the 'Geometries' BlockSourceReader, but as that can get simplified, perhaps this can too.